app/controllers/projects_controller.rb

```diff
@@ -188,6 +188,17 @@ def modules
     redirect_to action: 'settings', id: @project, tab: 'modules'
   end
 
+  def custom_fields
+    @project.work_package_custom_field_ids = params[:project][:work_package_custom_field_ids]
+    if @project.save
+      flash[:notice] = l(:notice_successful_update)
+    else
+      flash[:error] = l(:notice_project_cannot_update_custom_fields)
+    end
+
+    redirect_to action: 'settings', id: @project, tab: 'custom_fields'
+  end
```

Comment:
I fear I forgot to tell you that you also have to add the method to:

https://github.com/opf/openproject/blob/dev/config/initializers/permissions.rb#L48

Otherwise I alway receive a 403 upon trying to update.